python_ta/reporters/plain_reporter.py

```diff
@@ -112,8 +118,54 @@ def sort_messages(self):
         for msg in self._style_messages:
             self._sorted_style_messages[msg.msg_id].append(msg)
 
-    def show_file_linted(self, filename):
-        print('*'*15, 'File:', filename)
+    def set_output_stream(self, output_filepath_arg):
+        """Determine where to output pyta messages.
+        • Reset a file for outputting messages into, and store its file object.
+        • Default stream to std out.
+        • Note: leave the file open during the execution of pyta program 
+        because many writes may happen, and the fd should close automatically 
+        by the system when the program ends. 
+        • Raises IOError.
+        """
+        if output_filepath_arg is None and self._output_filepath is None:
+            # Stream to std out. Use method instead of setting self.out directly
+            self.set_output(sys.stdout)
+            return
+        elif output_filepath_arg is None:
+            return
+
+        # Paths may contain system-specific or relative syntax, e.g. `~`, `../`
+        correct_path = os.path.expanduser(output_filepath_arg)
+        if not os.path.exists(os.path.dirname(correct_path)):
+            raise IOError('path {} does not exist.'.format(output_filepath_arg))
+        if os.path.isdir(correct_path):
+            correct_path = os.path.join(correct_path, OUTPUT_FILENAME)
+
+        # Save output location and remove it if exists from previous run.
+        self._output_filepath = correct_path
+        # Remove existing file to prepare for appending messages from recursive 
+        # linting of files.
+        if os.path.exists(correct_path):
+            os.remove(correct_path)
+        
+        # Use this file object to append messages.
+        self.set_output(open(correct_path, 'a'))
+
+    def filename_to_display(self, filename):
+        """Display the file name, currently consistent with pylint format."""
+        return '{} File: {}'.format('*'*15, filename)
+
+    def register_file(self, filename):
+        """Display current filename to user. Also does some miscellaneous work 
+        with the file.
+        """
+        self.current_file_linted = filename
+        print(self.filename_to_display(filename), file=self.out)
```

Comment:
1. Don't print the filename here. Print it in `print_messages` as part of the report. That is, `print_messages` should be the only method that actually prints anything.
2. Both `set_output` and the `out` attribute are not necessary. Remove them.
3. `print_messages` should have a local variable that is actually the correct output stream. This way the call to `open` (if any) is as close as possible to the actual call to `print`.